src/simulations/collect.jl

```diff
@@ -0,0 +1,268 @@
+## Basic data collection functions
+
+get_data(a, s::Symbol) = getproperty(a, s)
+get_data(a, f::Function) = f(a)
+
+"""
+    collect_agent_data(model::ABM, properties::AbstractArray, step::Int)
+
+Collect agent properties into a dataframe. `properties` can have symbols (agent fields) or functions that take an agent as input.
+"""
+function collect_agent_data(model::ABM, properties::AbstractArray, step::Int)
+  dd = DataFrame()
```

Comment:
Since these functions are always for one `step`, why not return a [`DataFrameRow`](https://juliadata.github.io/DataFrames.jl/stable/lib/types/#DataFrames.DataFrameRow)?

Reply:
See [constructing row by row](https://juliadata.github.io/DataFrames.jl/stable/man/getting_started/#Constructing-Row-by-Row-1).

Reply:
Because they do not output a single row when you are collecting data from all agents.

Reply:
True. I'll think about that.

Reply:
`collect_model_data` does always outputs a single row. But its easier that both of these functions return the same object type.